Server/mods/deathmatch/logic/Utils.cpp

```diff
@@ -535,18 +535,104 @@ bool XMLColorToInt(const char* szColor, unsigned long& ulColor)
 
 bool XMLColorToInt(const char* szColor, unsigned char& ucRed, unsigned char& ucGreen, unsigned char& ucBlue, unsigned char& ucAlpha)
 {
-    // Convert it to an integer first
-    unsigned long ulColor;
-    if (!XMLColorToInt(szColor, ulColor))
-    {
+    // If we're empty, let's just stop right away
+    if (!szColor || strlen(szColor) == 0)
         return false;
+
+    std::vector<SColorRGBA> vecColors;
+    unsigned char           ucCount;
+
+    if (ColorStringToRGBA(szColor, SColorRGBA(ucRed, ucGreen, ucBlue, ucAlpha), vecColors, ucCount))
+    {
+        ucRed = vecColors[0].R;
+        ucGreen = vecColors[0].G;
+        ucBlue = vecColors[0].B;
+        ucAlpha = vecColors[0].A;
+
+        return true;
+    }
+
+    return false;
+}
+
+bool ColorStringToRGBA(const char* szColor, SColorRGBA defaultColor, std::vector<SColorRGBA>& vecColors, unsigned char& ucCount, bool bIgnoreAlpha)
+{
+    std::stringstream ss(szColor);
+    SColorRGBA        color = defaultColor;
+    bool              bPreviousWasHex = false;
+    unsigned int      uiRGBAIndex = 0;
+    unsigned long     ulColor;
+    unsigned char     ucValue;
+    unsigned char     ucLength = bIgnoreAlpha ? 3 : 4;
+
+    while (ss.good())
+    {
+        // Ambiguous value before a comma
+        SString strValue;
+        getline(ss, strValue, ',');
+
+        // Remove spaces
+        ReplaceOccurrencesInString(strValue, " ", "");
+
+        // Is the value looking like a hexadecimal?
+        if (strValue[0] == '#')

```

Comment:
can strValue be empty?

Reply:
> can strValue be empty?

Yes, but there is a catch.
If you enter `255, , #f00`, it would read it as palette colors: 255 (doesn't exist), 0 (black), red (255, 0, 0).

I could introduce a fix on line 587 for this so that you cannot mix palette and RGB colors:
```c++
ucCount += ucLength - uiRGBAIndex;
```

With this fix you can do the above, but it will force the colors to RGB if they aren't complete sets. So, above example would become:
`255, , #f00` -> red (255, 0, 0), red (255, 0, 0).

I suppose this fix is a good idea. What do you think?

Reply:
And if you enter an empty hexadecimal value it will error out in console:
`ERROR: Bad 'color' value specified in <vehicle> (line x)`.

Reply:
so strValue will always have at least one char?

Reply:
Actually you know what, the fix makes most sense since this function is shared across elements. Will fix.

Reply:
> so strValue will always have at least one char?

Yup.

Reply:
Now it's been changed so that empty values just skip as RGB. This also changes it so that you can't have both palette and RGB in the same color string.